install.sh

```diff
@@ -457,152 +456,56 @@ Mac system, you can use Vagrant or another virtualization tool. See our install
-  #
-  # The /proc/sys issue applies to Docker containers and probably other sorts of containers, too.
-  if egrep -q '/proc/sys\s+proc\s+ro' /proc/mounts || [ ! -e /etc/sysctl.conf ] ; then
-    fail "E_PROC_SYS_READ_ONLY" "You are using a Debian-derived Linux kernel, which needs a configuration option" \
-         "set in order to run Sandstorm. To set that option, please run the shell commands" \
-         "below as root. (If you are running this in a container through e.g. Docker, you will" \
-         "have to run the above commands _outside_ the container.)
-
-# sysctl -w kernel.unprivileged_userns_clone=1
-# echo 'kernel.unprivileged_userns_clone = 1' >> /etc/sysctl.conf"
-  fi
-
-  # OK, so it's a reasonable idea to ask the user to let us enable the sysctl.
-  USERNS_CLONE_UNPRIVILEGED_NEEDS_SYSCTL_SET=yes
-}
-
-enable_userns_sysctl_if_needed() {
-  # This function enables the Debian/Ubuntu-specific unprivileged
-  # userns sysctl.
-  #
-  # We only run it if we need to.
-  if [ "$USERNS_CLONE_UNPRIVILEGED_NEEDS_SYSCTL_SET" != "yes" ] ; then
+  if [ "no" = "$CURRENTLY_UID_ZERO" ] ; then
+    # Not root. Can't do anything about it.
     return
   fi
 
-  # It only makes sense when running as root, so if we are not currently running as root, just skip
-  # this code. If the user is not root, but they need this sysctl enabled, install.sh will
-  # eventually ask them for help escalating to root... hopefully.
-  if [ "no" = "$CURRENTLY_UID_ZERO" ] ; then
+  if [ ! -e /proc/sys/kernel/unprivileged_userns_clone ]; then
+    # No such sysctl on this system.
     return
   fi
 
-  local PRINT_USERNS_PROMPT="yes"
-  local ACCEPTED_SYSCTL_SWITCH="no"
-  local USE_SYSCTL_D="no"
-  local ENABLE_SYSCTL_STRING="(including in /etc/sysctl.conf)"
-  if [ -d /etc/sysctl.d ] ; then
-    # It's more polite to add drop-in files in /etc/sysctl.d than to modify /etc/sysctl.conf
-    # directly, so do that when possible.
-    USE_SYSCTL_D="yes"
-    ENABLE_SYSCTL_STRING="(including creating /etc/sysctl.d/50-sandstorm.conf)"
+  if [ "$(sysctl -n kernel.unprivileged_userns_clone)" = "1" ]; then
+    # Already enabled.
+    return
   fi
 
-  if [ "yes" = "${ACCEPTED_FULL_SERVER_INSTALL:-}" ] ; then
-    PRINT_USERNS_PROMPT="no"
-    ACCEPTED_SYSCTL_SWITCH="yes"
-  fi
+  # Enable it.
+  echo "NOTE: Enabling unprivileged user namespaces because you passed -d."
+  sysctl -wq kernel.unprivileged_userns_clone=1
 
-  if [ "${PRINT_USERNS_PROMPT}" = "yes" ] ; then
-    echo "Sandstorm requires sysctl kernel.unprivileged_userns_clone to be enabled."
-    echo "Currently, it is not enabled on your system."
-    if prompt-yesno "Shall I enable it for you ${ENABLE_SYSCTL_STRING}?" yes; then
-      ACCEPTED_SYSCTL_SWITCH="yes"
-    fi
+  # Also make sure it is re-enabled on boot. If sysctl.d exists, we drop our own config in there.
+  # Otherwise we edit sysctl.conf, but that's less polite.
+  local SYSCTL_FILENAME="/etc/sysctl.conf"
```

Comment:
Arguably feel free to bail on enabling the sysctl if `/etc/sysctl.d` doesn't exist. Not a show-stopper, just an opportunity for slightly fewer lines of code. vagrant-spk will have `/etc/sysctl.d`.